test/socket_test.dart

```diff
@@ -1,26 +1,176 @@
+    mockServer = await HttpServer.bind('localhost', 0);
+    mockServer.transform(WebSocketTransformer());
+  });
+
+  tearDown(() async {
+    if (mockServer != null) {
+      await mockServer.close();
+    } else {
+      print('mock server was null');
+    }
+  });
+
+  group('constructor', () {
+    test('sets defaults', () async {
+      final socket = Socket('wss://example.com/socket');
+      expect(socket.channels.length, 0);
+      expect(socket.sendBuffer.length, 0);
+      expect(socket.ref, 0);
+      expect(socket.endPoint, 'wss://example.com/socket/websocket');
+      expect(socket.stateChangeCallbacks, {
+        'open': [],
+        'close': [],
+        'error': [],
+        'message': [],
+      });
+      expect(socket.transport is IOWebSocketChannelClosure, true);
+      expect(socket.timeout, const Duration(milliseconds: 10000));
+      expect(socket.longpollerTimeout, 20000);
+      expect(socket.heartbeatIntervalMs, 30000);
+      expect(socket.logger is Logger, true);
+      expect(socket.reconnectAfterMs is Function, true);
+    });
+
+    test('overrides some defaults with options', () async {
+      final socket = Socket('wss://example.com/socket',
+          timeout: const Duration(milliseconds: 40000),
+          longpollerTimeout: 50000,
+          heartbeatIntervalMs: 60000,
+          logger: (kind, msg, data) => print('[$kind] $msg $data'));
+      expect(socket.channels.length, 0);
+      expect(socket.sendBuffer.length, 0);
+      expect(socket.ref, 0);
+      expect(socket.endPoint, 'wss://example.com/socket/websocket');
+      expect(socket.stateChangeCallbacks, {
+        'open': [],
+        'close': [],
+        'error': [],
+        'message': [],
+      });
+      expect(socket.transport is IOWebSocketChannelClosure, true);
+      expect(socket.timeout, const Duration(milliseconds: 40000));
+      expect(socket.longpollerTimeout, 50000);
+      expect(socket.heartbeatIntervalMs, 60000);
+      expect(socket.logger is Logger, true);
+      expect(socket.reconnectAfterMs is Function, true);
+    });
+  });
+
+  group('endpointURL', () {
+    test('returns endpoint for given full url', () {
+      final socket = Socket('wss://example.org/chat');
+      expect(
+        socket.endPointURL(),
+        'wss://example.org/chat/websocket?vsn=1.0.0',
+      );
+    });
+
+    test('returns endpoint with parameters', () {
+      final socket = Socket('ws://example.org/chat', params: {'foo': 'bar'});
+      expect(socket.endPointURL(),
+          'ws://example.org/chat/websocket?foo=bar&vsn=1.0.0');
+    });
+
+    test('returns endpoint with apikey', () {
+      final socket =
+          Socket('ws://example.org/chat', params: {'apikey': '123456789'});
+      expect(socket.endPointURL(),
+          'ws://example.org/chat/websocket?apikey=123456789&vsn=1.0.0');
+    });
+  });
+
+  group('connect with Websocket', () {
+    Socket socket;
+
+    setUp(() {
+      socket = Socket('wss://localhost:0/');
```

Comment:
your endpoint is wrong. it should be `'ws://localhost:${mockServer.port}'` to connect to mockServer.

However you can use this `'wss://localhost:0'` to check onError event trigger. It will return an WebSocketChannelException error.